server/services/party/party.emitter.ts

```diff
@@ -1,58 +1,19 @@
-import { ClientProxy } from '@nestjs/microservices'
-import { Inject, Injectable } from '@nestjs/common'
+import {ClientProxy}        from '@nestjs/microservices'
+import {Inject, Injectable} from '@nestjs/common'
 import {
-    PartyCreated,
-    GetPartyInvite,
-    PartyMemberJoined,
-    PartyMemberLeft,
-    MakePartyLeader,
-    PartyMemberKicked,
-} from './actions'
-import { WORLD_PREFIX } from '../world/world.prefix'
+    MadePartyLeader
+}                           from './actions'
+import {WORLD_PREFIX}       from "../world/world.prefix";
 
 @Injectable()
 export class PartyEmitter {
-    constructor(@Inject('WORLD_CLIENT') protected client: ClientProxy) {}
-
-    createParty(characterId: number, world: string) {
-        this.client.emit(
-            WORLD_PREFIX + PartyCreated.event,
-            new PartyCreated(world, characterId)
-        )
-    }
-
-    inviteToParty(leaderName: string) {
-        this.client.emit(
-            WORLD_PREFIX + GetPartyInvite.event,
-            new GetPartyInvite(leaderName)
-        )
-    }
-
-    joinParty(characterId: number, characterName: string, partyId: number) {
-        this.client.emit(
-            WORLD_PREFIX + PartyMemberJoined.event,
-            new PartyMemberJoined(characterId, characterName, partyId)
-        )
-    }
-
-    leaveParty(characterId: number, characterName: string, partyId: number) {
-        this.client.emit(
-            WORLD_PREFIX + PartyMemberLeft.event,
-            new PartyMemberLeft(characterId, characterName, partyId)
-        )
-    }
-
-    makeLeader(characterName: string) {
-        this.client.emit(
-            WORLD_PREFIX + MakePartyLeader.event,
-            new MakePartyLeader(characterName)
-        )
+    constructor(@Inject('WORLD_CLIENT') protected client: ClientProxy) {
     }
 
-    kickFromParty(characterId: number, partyId: number) {
+    madePartyLeader(partyId: number, characterId: number) {
```

Comment:
Changing the verb tense indicates it has already been done.
The emitter is like a results reporter so other micro services can do something based on the fact that this happened.

Reply:
Perhaps this exposes the problem that the naming used for these classes did not express its intent enough. I will look into renaming these classes to be easier to follow